pandas/tests/frame/methods/test_set_axis.py

```diff
@@ -98,3 +98,22 @@ class TestSeriesSetAxis(SharedSetAxisTests):
     def obj(self):
         ser = Series(np.arange(4), index=[1, 3, 5, 7], dtype="int64")
         return ser
+
+
+def test_nonkeyword_arguments_deprecation_warning():
+    # https://github.com/pandas-dev/pandas/issues/41485
+    df = DataFrame({"a": [1, 2, 3]})
+    msg = (
+        r"Starting with Pandas version 2\.0 all arguments of set_axis except for the "
+        r"arguments 'self' and 'labels' will be keyword-only"
+    )
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        df.set_axis([1, 2, 4], False)
```

Comment:
```suggestion
        df.set_axis([1, 2, 4], 0)
```

? (since it's for the axis keyword)

Also the same comment as in one of the other PRs, I would still assert the actual result to ensure we keep this working correctly until it's actually removed